File: archive/t/typescript/capitalize.ts
if (process.argv.length < 3 || process.argv[2].length < 1) {
    console.log("Usage: please provide a string")
    process.exit(0)
}

let myStr: string = process.argv[2];

const capitalize = (str: string) => str[0].toUpperCase() + str.slice(1);

console.log(capitalize(myStr));

File: archive/d/dart/capitalize.dart
import 'dart:io';

main(List<String> args) {

  if (args.isEmpty || args[0].isEmpty) {
    print("Usage: please provide a string");
    exit(1);
  }

  print(capitalize(args[0]));
}

String capitalize(String input) =>
    '${input[0].toUpperCase()}${input.substring(1)}';
